File: api/refresh_token.go
package main

import (
	"context"
	"encoding/json"
	"net/http"
	"universal-oauth/util"
)

func RefreshToken(w http.ResponseWriter, r *http.Request) {
	w.Header().Add("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)

	oauthConfig := util.GetOAuthConfig()
	code := r.URL.Query().Get("code")
	token, err := oauthConfig.Exchange(context.Background(), code)

	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	json.NewEncoder(w).Encode(token)
}

File: api/refresh_token.ts
import type {VercelRequest, VercelResponse} from "@vercel/node";
import {AuthorizationCode} from "simple-oauth2";
import {feedlyConfig} from "../util/util";

export default async (request: VercelRequest, response: VercelResponse) => {
  const client = new AuthorizationCode(feedlyConfig);
  const refreshToken = request.query.refresh_token as string;

  try {
    let accessToken = client.createToken({refresh_token: refreshToken});
    accessToken = await accessToken.refresh();
    response.setHeader('Content-Type', 'application/json');
    response.end(JSON.stringify(accessToken));
  } catch (error) {
    console.log('Refresh Token Error', error.message);
    response.end();
  }
};



